src/pytest_inmanta_lsm/remote_orchestrator.py

```diff
@@ -899,24 +914,73 @@ def wait_until_deployment_finishes(
-            assert (
-                resource["status"] == desired_state
-            ), f"Resource status do not match the desired state, got {resource['status']} (expected {desired_state})"
+        if not new_api:
+
+            def is_deployment_finished() -> bool:
+                response = self.client.get_version(self.environment, version)
+                assert response.result is not None
+                LOGGER.info(
+                    "Deployed %s of %s resources",
+                    response.result["model"]["done"],
+                    response.result["model"]["total"],
+                )
+                return response.result["model"]["total"] - response.result["model"]["done"] <= 0
+
+            retry_limited(is_deployment_finished, timeout)
+            result = self.client.get_version(self.environment, version)
+            assert result.result is not None
+            for resource in result.result["resources"]:
+                LOGGER.info(f"Resource Status:\n{resource['status']}\n{pformat(resource, width=140)}\n")
+                assert (
+                    resource["status"] == desired_state
+                ), f"Resource status do not match the desired state, got {resource['status']} (expected {desired_state})"
+        else:
+            self.wait_for_released(version)
+
+            def get_key_numbers():
+                result = self.client.resource_list(self.environment, deploy_summary=True, limit=1)
+                assert result.code == 200
+                summary = result.result["metadata"]["deploy_summary"]
+                # {'by_state': {'available': 3, 'cancelled': 0, 'deployed': 12, 'deploying': 0, 'failed': 0, 'skipped': 0,
+                #               'skipped_for_undefined': 0, 'unavailable': 0, 'undefined': 0}, 'total': 15}
+                return (
+                    (
+                        summary["by_state"]["deployed"]
+                        + summary["by_state"]["failed"]
+                        + summary["by_state"]["skipped"]
+                        + summary["by_state"]["skipped_for_undefined"]
+                        + summary["by_state"]["unavailable"]
+                        + summary["by_state"]["undefined"]
+                    ),
+                    summary["by_state"]["failed"],
+                    summary["total"],
+                )
+
+            def is_deployment_finished() -> bool:
+                done, failed, total = get_key_numbers()
+                LOGGER.info(
+                    "Deployed %s of %s resources",
+                    done,
+                    total,
+                )
+                return total - done <= 0
+
+            retry_limited(is_deployment_finished, timeout)
+            result = self.client.get_version(self.environment, version)
+
+            assert result.result is not None
+            # We only check first 1000
+            result = self.client.resource_list(self.environment, limit=1000)
```

Comment:
I think we should use paging here to get all resources, or at the very least, explain why we only check 1000 (if there is a reason).  We rarely have more than 1000 resources in testing, but for scaling test (where we might actually use this method) we do

Reply:
I agree, but I don't have the time to implement it. Could you add it?

Reply:
Paging is super annoying with the Client object, I created a ticket to improve support in core: https://github.com/inmanta/inmanta-core/issues/8801

In the meantime, I went for an alternative solution, we raise an exception after the first mismatching resource, so we can actually simply query the first mismatching resource from the api